src/Symfony/Component/Messenger/EventListener/SendFailedMessageToFailureTransportListener.php

```diff
@@ -27,12 +26,12 @@
  */
 class SendFailedMessageToFailureTransportListener implements EventSubscriberInterface
 {
-    private $failureSender;
+    private $failureSenders;
     private $logger;
 
-    public function __construct(SenderInterface $failureSender, LoggerInterface $logger = null)
+    public function __construct(array $failureSenders, LoggerInterface $logger = null)
```

Comment:
This is a BC-break. Can you make sure to also allow `SenderInterface`, which would be used by all receivers?

Reply:
What happens if there is no global failure transport, what should be the value of the SenderInterface?

`SendFailedMessageToFailureTransportListener.php` is instantiated only once. That means that this class is used by all transports to send messages to their failure transport (that can be a global or a local one defined in each transport).

Should I create one `SendFailedMessageToFailureTransportListener` for each failure transport? I would need an extra argument that tells me the transport name, so I can discard messages of different failed transports.

Reply:
My proposal, which I'm going to do next:

The constructor of `SendFailedMessageToFailureTransportListener`:
`public function __construct(?SenderInterface $globalfailureSender, LoggerInterface $logger = null, ServiceLocator $failureSenders)`

Will guarantee that BC is not broken.